javatools/src/main/java/org/xvm/tool/Launcher.java

```diff
@@ -156,6 +156,14 @@ public void run()
 
 
     // ----- text output and error handling --------------------------------------------------------
+    protected final void logAndAbort(boolean abort, Severity sev, String msg)
```

Comment:
The reason is that I always strive to preserve semantics, because I am not sure why something is designed the way it is. I was pretty sure the abort boolean set the exit flag to 1 or 0 depending on its value, and to preserve that, I propagated it into the LauncherException too, even though its mere presence on the catching side means that there was a non zero exit value.

Reply:
Btw - the abort boolean exists because displaying command line help does a System.exit(0). 

On the whole, the launchers cannot be hard coded to only ever be command line launchers, and warnings and errors need to be a programmatically accessible API, for language support to plug into IDEs and language servers and stuff. It's easy to turn a programmatically callable compiler, where you can access errors and warnings through an API into a command line tool. The other way around, turning a command line tool that was originally not meant to be anything else, into a programmatically invokable API, is much harder. Hence, I think we need to figure out some other things about the future of the javatools before adding a more generic fix. I'm just closing this then. This is the whole reason that I wanted to get a quick and not to awful fix in so that I could continue on the build system side without having to rerun anything in the debugger to figure out what is going on in the build, and not bump into anything more complex. The console is fine, but it's still just "screen scraping".